File: 2st/Riyeon/배열 만들기 4.java
import java.util.*;
class Solution {
    public int[] solution(int[] arr) {
        ArrayList<Integer> stk = new ArrayList<>();
        int i=0;
        while(i < arr.length){
            if(stk.isEmpty()){
                stk.add(arr[i]);
                i++;
            }
            else if(stk.get(stk.size()-1) < arr[i]){
                stk.add(arr[i]);
                i++;
            }
            else
                stk.remove(stk.size()-1);
        }
        
        return stk.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 2st/Yeonwoo/배열 만들기 4.py
def solution(arr):
    stk = []
    i=0
    while i < len(arr): 
        if stk==[]:
            stk.append(arr[i])
            i+=1
        elif stk!=[] and stk[-1] < arr[i]:
            stk.append(arr[i])
            i+=1
        elif stk!=[] and stk[-1] >= arr[i]:
            stk.remove(stk[-1])
    return stk